frontend/src/metabase/hooks/use-before-unload/use-before-unload.ts

```diff
@@ -0,0 +1,12 @@
+import { useBeforeUnload as useBeforeUnloadHook } from "react-use";
+import { t } from "ttag";
+
+// most browsers don't use a custom message with beforeunload anymore, just putting here to retain compatibility
+// https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#compatibility_notes
+export const BEFORE_UNLOAD_UNSAVED_MESSAGE = t`You have unsaved changes.`;
+
+const useBeforeUnload = (condition: (() => boolean) | boolean) => {
```

Comment:
nit: it can be too much, but consider this 😄 

```diff
-const useBeforeUnload = (condition: (() => boolean) | boolean) => {
+type Condition = Parameters<typeof useBeforeUnloadHook>[0]
+const useBeforeUnload = (condition: Condition) => {
   return useBeforeUnloadHook(condition, BEFORE_UNLOAD_UNSAVED_MESSAGE);
 };
 ```

Reply:
More typescript is (almost) always good 😄